src/stores/HomeScreenStore.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+
+import type { Coordinates, NearbyLocation } from '../types';
+import type { IComponentStore } from './types';
+
+import { action, computed, observable, reaction, runInAction } from 'mobx';
+import debounce from 'lodash.debounce';
+import { LoadObject } from 'brewskey.js-api';
+import GoogleApi from '../GoogleApi';
+import NearbyLocationsStore from '../stores/NearbyLocationsStore';
+import { getCurrentGPSCoordinates } from '../utils';
+
+const SEARCH_TEXT_DEBOUNCE_TIMEOUT = 1000;
+
+class HomeScreenStore implements IComponentStore {
+  _nearbyLocationsStore = new NearbyLocationsStore();
+
+  _changeSearchTextReactionDispose: () => void;
+  _closeSearchBarReactionDispose: () => void;
+  _gspLoaderReactonDispose: () => void;
+  _searchCoordinatesReactionDispose: () => void;
+
+  @observable isSearchBarVisible: boolean = false;
+  @observable searchText: string = '';
+  @observable
+  _searchCoordinatesLoader: LoadObject<Coordinates> = LoadObject.empty();
```

Comment:
You only really need one set of coordinates. You can basically use this logic to check what store you should fetch the coordinates from.

```
if (this.isSearchBarVisible) {
  if (!this.searchText) {
    return LoadObject.empty();
  }
  return GoogleApi.fetchCoordinatesByAddress(
    this.searchText,
  );
} else {
  return getCurrentGPSCoordinatesLoader();
}
```

Reply:
Well, may be If I'll find a way to make this ` getCurrentGPSCoordinatesLoader();` working, read another comment about it.
the first if {} logic should be debounced, so not sure if all this will end up much simpler anyways.